beacon_chain/rpc/rest_validator_api.nim

```
@@ -418,17 +410,40 @@ proc installValidatorApiHandlers*(router: var RestRouter, node: BeaconNode) =
             return RestApiResponse.jsonError(Http400, InvalidSlotValueError)
         if res.isErr():
           return RestApiResponse.jsonError(Http400, res.error())
-        res.get.blck
+        res.get
     return
-      if contentType == sszMediaType:
-        let headers = [("eth-consensus-version", message.kind.toString())]
-        withBlck(message):
+      withBlck(message.blck):
+        let data =
+          when blck is deneb.BeaconBlock:
```

This condition in isolation is fine -- just as well not to assume what the E-fork `produceBeaconBlockV2` will look like. But right now, it will silently fall back to the pre-Deneb case.

This is another use case for using something similar to a `static: doAssert` or `static: raiseAssert` in an `else` branch of the `when` to simulate the kind of exhaustiveness checking one can get from `case` at runtime. While right now, this indeed covers all forks, it becomes a bit of a maintenance time-hazard because it'll have to be found manually as-is, rather than failing at compile time visibly/obviously when the next fork support gets added.

Otherwise, looks good.

---

9df97f8d70fa69f3635404b594d4881df60044ac